spec/haskell/src/SEL4/Object/SchedContext.lhs

```diff
@@ -525,6 +525,7 @@ This module uses the C preprocessor to select a target architecture.
 > schedContextZeroRefillMax :: PPtr SchedContext -> Kernel ()
 > schedContextZeroRefillMax scPtr = do
 >     updateSchedContext scPtr $ (\sc -> sc { scRefillMax = 0 })
+>     updateSchedContext scPtr $ (\sc -> sc { scSporadic = False })
```

Comment:
This is a bit odd, we now have `schedContextZeroRefillMax` setting `scSporadic`, which minimises turbulence in the rest of the proof, but doesn't seem to make a lot of sense to read. This file is a bit sparse on commentary, which doesn't make it easier to grasp the meaning for the uninitiated.

Reply:
This function has drifted over the last few years, so the name is no longer a good one. This function is now probably better thought of now as zeroing (in a general sense) all of the fields for the "sporadic" parts of the sched context. But I'm not really sure what "sporadic" means any more...

Would it be better to call it `sched_context_zero_sporadic_fields` or something? I can't think of anything better

Reply:
Yeah, your comment and the better name makes more sense (I'm guessing you'd want a `camelCase` name since this is Haskell). If it's inconvenient to do in this PR, can maybe add a rename to another day, but the helpful comment would still be welcome.

Reply:
Actually, maybe `sched_context_zero_sporadic_fields` isn't such a good name: presumably we could have the `sc_sporadic` field be `False`, so that sched context isn't "sporadic", but we'd still like to call this function to make the sc inactive. So it really does seems like it makes it inactive (which is why it was called `zero_refill_max` to begin with) and then we've just got a bunch of other things we'd like to zero out in order to keep the invariants happy. Still not sure what to call it, then.

I think it should be renamed, but considering that the name was off before this work was started, it would be better to redo the naming in another PR.

And yes, `camelCase` for the Haskell, but we do have the analogue of this function in the abstract spec as well.

Reply:
I think making it inactive is the right view -- that requires resetting the refills and the sporadic flag at the same time, because it's not properly inactive if the sporadic flag is still set.